x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -242,9 +242,6 @@ private RoleReferenceIntersection buildRoleReferencesForRemoteAccess() {
             // access of the overall intersection accordingly
             roleReferences.add(new RoleReference.RemoteAccessRoleReference(RemoteAccessAuthentication.RoleDescriptorsBytes.EMPTY));
         } else {
-            // TODO handle this once we support API keys as querying subjects
```

Comment:
This part touches the fulfilling cluster side -- I'd reserve these for a separate PR. The code changes may be quite small (i.e., just deleting assertions) but we'll want to add tests which could add quite a lot to PR size.

Reply:
👍 I'm happy to address it in a separate PR.